app/v3/controllers/SubmitSelfEmploymentBsasController.scala

```diff
@@ -73,11 +73,9 @@ class SubmitSelfEmploymentBsasController @Inject()(val authService: EnrolmentsAu
             //Submit Return to ETMP
             EitherT(service.submitSelfEmploymentBsas(parsedRequest))
           }
-          hateoasResponse <- EitherT.fromEither[Future](
-            hateoasFactory
-              .wrap(response.responseData, SubmitSelfEmploymentBsasHateoasData(nino, calculationId))
-              .asRight[ErrorWrapper])
         } yield {
+          val hateoasResponse = hateoasFactory.wrap(response.responseData, SubmitSelfEmploymentBsasHateoasData(nino, calculationId, None))
```

Comment:
ah I can see sometimes we are using the val hateoasData, hateoasResponse and sometime we are not.

Happy to leave it as it is if you want